### PR TITLE
docs: update v2.3+ roadmap

### PR DESCRIPTION
**Pull request by alexmt**
Signed-off-by: Alexander Matyushentsev <AMatyushentsev@gmail.com>

PR adds a set of new features to v2.3+ roadmap and removes cluster browsing UI since we don't plan to work on it soon.

@jessesuen , @jannfis , @sbose78 would you like to hold off adding something or add anything else? E.g. add `Merge Image updater into Argo CD` ?

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -44,17 +45,22 @@ Instead of asking an administrator to change Argo CD settings end users can perf
 
 ## v2.3 and beyond
 
-### Application Details Page Usability
+### Input Forms UI Refresh
+
+Improved design of the input forms in Argo CD Web UI: https://www.figma.com/file/IIlsFqqmM5UhqMVul9fQNq/Argo-CD?node-id=0%3A1
+
+### Merge ApplicationSet controller into Argo CD
```
Comment (sbose78) [inline docs/roadmap.md:52]: 👍 

Minimally, move the repo. Cautiously, move the controller into the same process.

File: docs/roadmap.md [hunk 2]
```diff
@@ -44,17 +45,22 @@ Instead of asking an administrator to change Argo CD settings end users can perf
...
+The ApplicationSet functionality is available in Argo CD out-of-the-box ([#7351](https://github.com/argoproj/argo-cd/issues/7351)).
+The Argo CD UI/CLI/API allows to manage ApplicationSet resources same as Argo CD Applications ([#7352](https://github.com/argoproj/argo-cd/issues/7352)).
+
+### Merge Argo CD Notifications into Argo CD
 
-Application details page has accumulated multiple usability and feature requests such as 
-[Node view](https://github.com/argoproj/argo-cd/issues/1483),
-Network view ([1](https://github.com/argoproj/argo-cd/issues/2892), [2](https://github.com/argoproj/argo-cd/issues/2338))
- [etc](https://github.com/argoproj/argo-cd/issues/2199).
+The [Argo CD Notifications](https://github.com/argoproj-labs/argocd-notifications) feature is now part of Argo CD: [#7350](https://github.com/argoproj/argo-cd/issues/7350)
```
Comment (sbose78) [inline docs/roadmap.md:59]: 👍

**Review by sbose78 — APPROVED** [review]

**Comment by jannfis** [comment]
I'd imho like to see multi-tenancy improvements on the roadmap as well. I still have not received any feedback on my [proposal](https://github.com/argoproj/argo-cd/pull/6409)

**Comment by jannfis** [comment]
Regarding the Image Updater, there's quite a few things to do before being in a state that could be merged into the main Argo CD component. Most notably, I want it to be controlled by its own API (https://github.com/argoproj-labs/argocd-image-updater/issues/235). Long term, integration into Argo CD would possibly make much sense. But yeah, if the roadmap is "2.3 and beyond", then totally, I'm all for this feature to be an integrated experience.

**Comment by alexmt** [comment]
Thank you for review @sbose78 and @jannfis 

> Regarding the Image Updater, there's quite a few things to do before being in a state that could be merged into the main Argo CD component.

Created the issue https://github.com/argoproj/argo-cd/issues/7385 and added it to the roadmap. Called out that image updater will be ready to get merged then v1.0 version is published.

> I'd imho like to see multi-tenancy improvements on the roadmap as well. I still have not received any feedback on my proposal

Really sorry it took forever to get this proposal moving. We discussed it with @sbose78 during one of the contributors' meetings (https://www.youtube.com/watch?v=fiJq8kKrzEE&ab_channel=ArgoProj  recordings are useful sometimes :)  ) We've agreed that both use cases (app source and full app in namespace) are useful and should be supported. I proposed that we should try to implement Applications outside argocd namespace as a labs project , probably as part of https://github.com/argoproj-labs/appsource. 100% agree it should be on roadmap - added to the doc.

**Review by jessesuen — APPROVED** [review]
I agree with a cautious approach to moving these things into Argo core, starting with the move out of argoproj-labs to argoproj. Some of these may always be optional components, and others we may decide they warrant full integration as part of base installation.

**Pull request merged**